packages/safe-apps-sdk/src/communication/index.ts

```diff
@@ -52,16 +52,21 @@ class PostMessageCommunicator implements Communicator {
     }
   };
 
-  public send = <M extends Methods, P, R>(method: M, params: P): Promise<Response<R>> => {
+  public send = <M extends Methods, P, R>(method: M, params: P): Promise<SuccessResponse<R>> => {
     const request = MessageFormatter.makeRequest(method, params);
 
     if (typeof window === 'undefined') {
       throw new Error("Window doesn't exist");
     }
 
     window.parent.postMessage(request, '*');
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       this.callbacks.set(request.id, (response: Response<R>) => {
+        if (!response.success) {
+          reject(new Error(response.error));
```

Comment:
What are some examples of these errors? Maybe we should prefix them with something like "SDK error"?

Reply:
If there's a problem processing app's request on the interface side it will send back a message with `success: false`. One I can think of a 404 from the server for `sdk.txs.getTxBySafeTxHash`